frontend/src/components/InvalidEvents.vue

```diff
@@ -0,0 +1,238 @@
+              <span v-if="item.type === 'context'">Context</span>
+            </template>
+
+            <template v-slot:item.name="{ item }">
+              <span>{{ item.name }}</span>
+            </template>
+            <template v-slot:item.createdAt="{ item }">
+              <span>{{ formatDate(item.createdAt) }}</span>
+            </template>
+            <template v-slot:no-data>No Invalid Events Available</template>
+          </v-data-table>
+        </v-card>
+      </v-col>
+    </v-row>
+
+    <v-dialog v-model="dialog" max-width="700px" closeable>
+      <v-card>
+        <v-card-title>
+          <span class="headline">Event details</span>
+        </v-card-title>
+        <v-card-text>
+          <h4>{{ detailView.displayName }}</h4>
+          <v-card outlined class="pa-2">
+            <pre style="overflow: auto; display: grid">
+              <h4 v-if="detailView.displaySchema.valid">Valid</h4>
+              <p v-if="detailView.displaySchema.valid" style="color: green">{{ detailView.displaySchema.valid }}</p>
+              <h4 v-if="detailView.displaySchema.invalid">Wrong value</h4>
+              <p v-if="detailView.displaySchema.invalid" style="color: red">{{ detailView.displaySchema.invalid }}</p>
+              <h4 v-if="detailView.displaySchema.notSet">Missing</h4>
+              <p v-if="detailView.displaySchema.notSet" style="color: grey">{{ detailView.displaySchema.notSet }}</p>
+            </pre>
+          </v-card>
+        </v-card-text>
+      </v-card>
+    </v-dialog>
+  </v-container>
+</template>
+
+<script>
+import Help from './Help'
+import invalidEventClient from '../lib/rest/invalidEvents'
+import eventClient from '../lib/rest/events'
+import contextClient from '../lib/rest/contexts'
+import moment from 'moment'
+
+export default {
+  components: { Help },
+  data() {
+    return {
+      dialog: false,
+      event: {},
+      detailView: {
+        displayName: '',
+        displaySchema: {
+          valid: {},
+          invalid: {},
+          notSet: {}
+        }
+      },
+      detailErrors: {},
+      eventSpecs: [],
+      headers: [
+        { text: 'Source', value: 'type' },
+        {
+          text: 'Event name',
+          align: 'left',
+          value: 'name'
+        },
+        { text: 'Created', value: 'createdAt' },
+        { text: 'Actions', value: 'action', align: 'right', sortable: false }
+      ],
+      valid: false,
+      rules: {
+        required: (value) => !!value || 'Required.'
+      }
+    }
+  },
+  methods: {
+    remove(eventSpec) {
+      invalidEventClient.delete(eventSpec.id)
+
+      let idx = this.eventSpecs.findIndex((es) => es.id === eventSpec.id)
+      this.eventSpecs.splice(idx, 1)
+    },
+    details(event) {
+      this.event = event
+
+      const json = event.type === 'properties' ? event.properties : event.context
+
+      if (event.type === 'properties') {
+        try {
+          const properties = eventClient.list()
```

Comment:
I think you could remove the variable assignment and directly write `eventClient.list().then()`